src/components/App/App.test.js

```diff
@@ -1,34 +1,97 @@
 import React from 'react';
-import { getNodeText, render } from '@testing-library/react';
+import { fireEvent, getNodeText, render } from '@testing-library/react';
 
 import App from './App';
 
 describe('App', () => {
-  let instance;
-  beforeEach(() => {
-    instance = render(<App />);
-  });
+  describe('displays all expected parts', () => {
+    let instance;
+    beforeEach(() => {
+      instance = render(<App />);
+    });
 
-  it('renders the title', () => {
-    const title = instance.getAllByRole('heading')
-      .map((heading) => getNodeText(heading))
-      .find((text) => text === 'Shopping List');
-    expect(title).toBeDefined();
-  });
+    it('renders the title', () => {
+      const title = instance.getAllByRole('heading')
+        .map((heading) => getNodeText(heading))
+        .find((text) => text === 'Shopping List');
+      expect(title).toBeDefined();
+    });
 
-  it('renders the number of items on the list', () => {
-    expect(instance.getByText(/[0-9]+ items/)).toBeInTheDocument();
-  });
+    it('renders the number of items on the list', () => {
+      expect(instance.getByText(/[0-9]+ items/)).toBeInTheDocument();
+    });
 
-  it('renders the logo', () => {
-    expect(instance.getByTitle('Logo')).toBeInTheDocument();
-  });
+    it('renders the logo', () => {
+      expect(instance.getByTitle('Logo')).toBeInTheDocument();
+    });
+
+    it('renders the share icon', () => {
+      expect(instance.getByLabelText('Share this list')).toBeInTheDocument();
+    });
 
-  it('renders the share icon', () => {
-    expect(instance.getByLabelText('Share this list')).toBeInTheDocument();
+    it('renders the box to enter new items', () => {
+      expect(instance.getByPlaceholderText('Item Name')).toBeInTheDocument();
+    });
   });
 
-  it('renders the box to enter new items', () => {
-    expect(instance.getByPlaceholderText('Item Name')).toBeInTheDocument();
+  describe('when an item is added to the list', () => {
```

Comment:
These tests are moved out of ShoppingList to here since they are now more applicable here.

Reply:
Looks good, although I think the test 'renders the box to enter new items' now needs the placeholder changing to "Add something to the list..." as I made the placeholder more generic in the UI updates. My fault obviously, but I don't think the test will pass without the change! There may be a couple of instance where it's referred to actually.